cmd/dispatcher/main.go

```diff
@@ -110,6 +110,9 @@ func main() {
 			if errors.Is(err, context.Canceled) {
 				return
 			}
+			rmqHelper.CloseRabbitMQConnections(env.connection, logger)
+			time.Sleep(time.Second)
+			rmqHelper.SignalRetry(true)
 		}
 		logger.Warn("recreating RabbitMQ resources")
 		if retry := rmqHelper.WaitForRetrySignal(); !retry {
```

Comment:
This got me thinking if this code is ever exercised. The only time the retryChannel gets a false (for dispatcher) is when `CleanupRabbitMQ` is called. And that's in a defer which will run after this function exits.

Reply:
just when connection creation errors, but yep it seems to be missing a case when consume from queue fails